### PR TITLE
Try weaning ourselves completely off of `staticfloat/julia*` Homebrew taps

### PR DESCRIPTION
**Pull request by staticfloat**
This is a shot at fixing https://github.com/JuliaLang/julia/issues/30949.  Note that this will cause us to start building SuiteSparse again, but that's not that bad; SS builds pretty quickly.  The most important two are LLVM and OpenBLAS, which are both covered by BB at this point, so we don't need to use my outdated tap anymore.

### CODE REVIEW

**Comment by ViralBShah** [comment]
We also have a SuiteSparseBuilder. Time to move to using it:
https://github.com/JuliaLinearAlgebra/SuiteSparseBuilder

And also openlibmbuilder?
https://github.com/JuliaMath/OpenlibmBuilder

**Comment by vtjnash** [comment]
> ld: library not found for -lopenblas
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> make[2]: *** [/Users/travis/build/JuliaLang/julia/usr/manifest/suitesparse] Error 1

**Comment by staticfloat** [comment]
@andreasnoack @ViralBShah As part of integrating BB-installed SuiteSparse I have attempted to reduce differences between [the BB installation script](https://github.com/JuliaPackaging/Yggdrasil/blob/9f309cecb94d8b72bce35ec0ef9cc9b10f288d92/SuiteSparse/build_tarballs.jl) and this one.  The two big ones are:

* We don't use `make library` but instead do `for subproj in <list of sub projects>; do make -C ${subproj} library; done`.  This avoids trying to build things like GraphBLAS that we don't care about.
* I have set `-DNPARTITION` once again.  That wasn't set when building SS 4.4.6, but I'm not sure if that was on purpose or not; it seems to me that without setting `-DNPARTITION` we would require `libmetis`?

Please let me know if I've done anything wrong that will screw up the sparse landscape.

**Comment by ViralBShah** [comment]
`-DNPARTITION` is right. I don't think we ever distributed Metis. However, given that Metis is now apache licensed, and we have BB, it would be nice to have going forward.

We should update to the latest suitesparse - maybe first get this working as is, and then bump, once the BB stuff is stable.

For list of subprojects, it seems we are doing the following:
`for proj in AMD BTF CAMD CCOLAMD COLAMD CHOLMOD LDL KLU UMFPACK RBio SPQR`

I believe we do not have bindings for KLU, LDL or RBio. @andreasnoack Do you know if we can safely ignore those?

**Comment by staticfloat** [comment]
Do we want to disable OpenMP?  [We do so on Windows](https://github.com/JuliaPackaging/Yggdrasil/blob/b9452926cd06d0af4544d35c14b3c1c404ce62fa/SuiteSparse/build_tarballs.jl#L22) in BB-land, and the OpenMP flag is causing a problem on the FreeBSD CI bot (where I assume OpenMP is just not installed; it is installed on BB).  I remember back in the mists of time we disabled OpenMP because we wanted a different way of controlling parallelism, but I'm not sure if that is purposeful or not.

Unless I hear otherwise, I am going to disable OpenMP in the from-source builds here just so that we can get CI passing again.  Note that while we have BB tarballs available for all platforms, we're only really using it on Travis OSX and Appveyor, presumably because we want to continue testing building from source on _some_ of the platforms.

**Comment by staticfloat** [comment]
Okay, shortlist of changes within this branch:

* Bump SuiteSparse version to 5.4.0, change installation parameters somewhat (add `-DNPARTITION` and disable `-fopenmp` when building from-source)
* Add BB installation for SuiteSparse and OpenLibm, turn on both for OSX Travis, turn on only SuiteSparse for Appveyor, since it seems there's some special OpenLibm handling there, not sure what to do about it.

If this passes CI and the build changes are acceptable to the Sparsity Shifu's, let's merge it.

File: .travis.yml [hunk 1]
```diff
@@ -73,23 +73,18 @@ before_install:
...
-        brew rm --force $(brew deps --HEAD julia);
-        brew install -v ccache gcc gmp mpfr pcre2 staticfloat/julia/openblas-julia staticfloat/julia/suite-sparse-julia staticfloat/juliadeps/libgfortran;
-        BUILDOPTS="-j3 USECLANG=1 USECCACHE=1 USE_BINARYBUILDER_LLVM=1 USE_BINARYBUILDER_OPENBLAS=1 BINARYBUILDER_LLVM_ASSERTS=1";
-        BUILDOPTS="$BUILDOPTS LLVM_CONFIG=$TRAVIS_BUILD_DIR/usr/tools/llvm-config LLVM_SIZE=$TRAVIS_BUILD_DIR/usr/tools/llvm-size";
-        BUILDOPTS="$BUILDOPTS VERBOSE=1 USE_BLAS64=0 SUITESPARSE_INC=-I$(brew --prefix suite-sparse-julia)/include FORCE_ASSERTIONS=1";
-        BUILDOPTS="$BUILDOPTS LIBBLAS=-lopenblas LIBBLASNAME=libopenblas LIBLAPACK=-lopenblas LIBLAPACKNAME=libopenblas";
-        for lib in SUITESPARSE BLAS LAPACK GMP MPFR LIBUNWIND; do
+        brew rm --force gcc gmp mpfr pcre2;
+        brew install -v gcc gmp mpfr pcre2;
+        BUILDOPTS="-j3 USECLANG=1 USECCACHE=1 VERBOS=1 FORCE_ASSERTIONS=1";
```
Comment (fredrikekre) [inline .travis.yml]:
```suggestion
        BUILDOPTS="-j3 USECLANG=1 USECCACHE=1 VERBOSE=1 FORCE_ASSERTIONS=1";
```

**Comment by staticfloat** [comment]
Looks like this is good to go; the 32-bit windows build timed out, but I'm trying it again.

**Comment by StefanKarpinski** [comment]
🎉

**Pull request merged**